[JIT] Trace module calls

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29261 [JIT] Trace module calls**
* #28864 [JIT] API for finding a common ancestor block for a pair of nodes

Differential Revision: [D18343363](https://our.internmc.facebook.com/intern/diff/D18343363)

Commit:
[WIP][JIT] Trace module calls

[ghstack-poisoned]

Commit:
Update on "[WIP][JIT] Trace module calls"




[ghstack-poisoned]

Commit:
Update on "[WIP][JIT] Trace module calls"


Differential Revision: [D18343363](https://our.internmc.facebook.com/intern/diff/D18343363)

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/passes/fixup_trace_scope_blocks.cpp:
```diff
@@ -0,0 +1,440 @@
+  convertReturnsToTuples(graph->block());
+  if (!self) {
+    // We have no Module, so we're just going to inline everything.
+    // This should give us a totally flag graph.
```
flag -> flat

zdevito reviewed (approved):
This looks great! Only minor asks for comments.

Commit:
Update on "[JIT] Trace module calls"


Differential Revision: [D18343363](https://our.internmc.facebook.com/intern/diff/D18343363)

[ghstack-poisoned]
